core/services/ocr2/validate/validate.go

```diff
@@ -178,11 +184,57 @@ func validateOCR2GenericPluginSpec(jsonConfig job.JSONConfig) error {
 		return errors.New("generic config invalid: must provide plugin name")
 	}
 
-	if p.TelemetryType == "" {
-		return errors.New("generic config invalid: must provide telemetry type")
+	if p.OCRVersion != 2 && p.OCRVersion != 3 {
+		return errors.New("generic config invalid: only OCR version 2 and 3 are supported")
 	}
 
-	return nil
+	plugEnv := env.NewPlugin(p.PluginName)
+
+	command := p.Command
+	if command == "" {
+		command = plugEnv.Cmd.Get()
+	}
+
+	if command == "" {
+		return errors.New("generic config invalid: no command found")
+	}
+
+	_, err = exec.LookPath(command)
+	if err != nil {
+		return fmt.Errorf("failed to find binary  %q", command)
+	}
+
+	envVars, err := plugins.ParseEnvFile(plugEnv.Env.Get())
+	if err != nil {
+		return fmt.Errorf("failed to parse median env file: %w", err)
```

Comment:
why`median env file`? typo?

Reply:
Fixed in https://github.com/smartcontractkit/chainlink/pull/12430/commits/2cfe4e4666ed312664416bc92da06659e548f357